contracts/FluidityTap.sol

```diff
@@ -53,6 +53,17 @@ contract FluidityTap is ERC20Detailed, ERC20Burnable, ERC20Mintable, Ownable, Wh
         addWhitelisted(msg.sender); // ensure owner is whitelisted
     }
 
+    /**
+      * @notice Sets a new Treasury
+      * @dev Can only be performed by owner.
+      *
+      */
+    function updateTreasury(bytes9 _newCusip,  uint256 _newTotalFaceValue, uint256 _newMaturityDate) external onlyOwner {
+        cusip = _newCusip;
+        totalFaceValue = _newTotalFaceValue;
+        maturityDate = _newMaturityDate;
```

Comment:
Should there be any checks on the value of the maturity date? e.g. can it be 0? should it always be in the future?

Reply:
no, it can and may start out as zero. different treasuries depending on which one is purchased can have variable due dates, ideally likely increasing, but not seen as a hard requirement.